bundles/org.openhab.binding.blink/src/main/java/org/openhab/binding/blink/internal/handler/AccountHandler.java

```diff
@@ -0,0 +1,402 @@
+        // the framework is then able to reuse the resources from the thing handler initialization.
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Wait for login");
+
+        scheduler.execute(() -> {
+
+            // register 2FA Verification Servlet for this thing
+            if (accountServlet == null) {
+                try {
+                    logger.debug("Registering 2FA servlet");
+                    accountServlet = new AccountVerificationServlet(httpService, bundleContext, this, blinkService);
+                } catch (IllegalStateException e) {
+                    logger.warn("Failed to create account servlet", e);
+                }
+            }
+
+            // get (or generate for new bridges) the blink client id
+            boolean start2FA = false;
+            Map<String, String> properties = editProperties();
+            String generatedClientId = properties.get(GENERATED_CLIENT_ID);
+            if (generatedClientId == null) {
+                logger.debug("Generating new Client ID, starting new 2FA process");
+                generatedClientId = blinkService.generateClientId();
+                start2FA = true;
+                properties.put(GENERATED_CLIENT_ID, generatedClientId);
+            }
+            this.generatedClientId = generatedClientId;
+            try {
+                logger.debug("Logging in to blink servers with generated client id {}", generatedClientId);
+                // call login api
+                blinkAccount = blinkService.login(config, generatedClientId, start2FA);
+                logger.debug("Successfully logged in with account {}", blinkAccount);
+                blinkAccount.lastTokenRefresh = Instant.now();
+                properties.putAll(blinkAccount.toAccountProperties());
+                // don't know how to get scheme and port from openhab, right now it's hardcoded...
+                String scheme = "http://";
+                int port = 8080;
+                String validationUrl = scheme + networkAddressService.getPrimaryIpv4HostAddress() + ":" + port
+                        + "/blink/" + thing.getUID().getId();
+                properties.put("validationUrl", validationUrl);
+                logger.debug("Updating bridge properties");
+                updateProperties(properties);
+                // do 2FA if necessary
+                if (blinkAccount.account.client_verification_required) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "Waiting for 2 Factor Authentication. Please go to " + validationUrl
+                                    + " to enter the PIN you received via SMS or email");
+                } else {
+                    setOnline();
+                }
+            } catch (Exception e) {
```

Comment:
You should specify the concrete exception you want to catch, since catching `Exception` catches also all runtime exceptions.